plugins/ci/skills/github-app-auth/github-app-auth.sh

```
@@ -0,0 +1,84 @@
+  }
+  jwt="${header}.${payload}.${signature}"
+
+  # --- exchange JWT for installation token ---
+  local response token
+  response=$(curl -fsS -X POST \
+    --connect-timeout 10 --max-time 30 \
+    -H "Authorization: Bearer ${jwt}" \
+    -H "Accept: application/vnd.github+json" \
+    "https://api.github.com/app/installations/${install_id}/access_tokens" 2>&1)
+
+  if [[ $? -ne 0 ]]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)